src/main/java/com/box/sdk/internal/utils/CollectionUtils.java

```diff
@@ -0,0 +1,65 @@
+package com.box.sdk.internal.utils;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * {@link Collection} related utlities.
+ *
+ * @author Stanislav Dvorscak
+ *
+ */
+public class CollectionUtils {
+
+    /**
+     * Only static members.
+     */
+    protected CollectionUtils() {
+    }
+
+    /**
+     * Re-maps a provided collection.
+     *
+     * @param <T_Result>
+     *            type of result
+     * @param <T_Source>
+     *            type of source
+     * @param source
+     *            for mapping
+     * @param mapper
+     *            element mapper
+     * @return mapped source
+     */
+    public static <T_Result, T_Source> List<T_Result> map(Collection<T_Source> source, Mapper<T_Result, T_Source> mapper) {
```

Comment:
I'm not sure about this method. It really isn't doing much, and an inline foreach loop is much more readable in my opinion.